pkg/credentialprovider/config.go

```diff
@@ -64,6 +74,32 @@ func GetPreferredDockercfgPath() string {
 }
 
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	// Try happy path first - latest config file
```

Comment:
[https://github.com/docker/docker/blob/master/cliconfig/config.go#L159](https://github.com/docker/docker/blob/master/cliconfig/config.go#L159)
In `func Load(configDir string) (*ConfigFile, error)`, first, they try to load new configuration file at `~/.docker/config.json` and parse by `configFile.LoadFromReader(file)`, if the config file is not found, then they fallback to load the old one at `~/.dockercfg` by `configFile.LegacyLoadFromReader(file)`.